theia-turn/turn-sprotty/src/views/ucm/connection.edge.tsx

```diff
@@ -30,16 +30,20 @@ export class ConnectionEdgeView implements IView {
         } else if (segments.length > 2) {
             const p2 = segments[2];
             const p3 = segments[3];
-            const c1 = p1.x - p0.x;
-            const c2 = p3.x - p1.x;
+            const midLength = Math.abs(p1.y - p2.y);
+            const c = Math.min(midLength / 2.0, p1.x - p0.x, p3.x - p1.x);
```

Comment:
Any particular reason for halving our delta y?

Reply:
Yes my good fellow! So my thought process for the curved node connection issue: 3 lines create the complete route of certain node connections, given from the following points:
[p0.x < p1.x = p2.x < p3.x]
[p0.y = p1.y < p2.y = p3.y ] or [p0.y = p1.y > p2.y = p3.y]
The "midlength" variable is the middle vertical line where p1.x = p2.x. 
The idea is that the curve (arc radius) should be determined by the minimum of the following values:
- the length of the middle vertical line divided by 2
- the length of the first line from p0 to p1 (p1.x - p0.x)
- the length of the last line from p2 to p3 (p3.x - p1.x)
This will prevent any weird paths from appearing.